pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -65,6 +67,10 @@ func (cfg *Config) Validate() error {
 		if err := cfg.ResultsCacheConfig.Validate(); err != nil {
 			return errors.Wrap(err, "invalid ResultsCache config")
 		}
+	} else {
+		if cfg.MaxSeriesPerShard > 0 {
+			return errors.New("-query-frontend.query-sharding-max-series-per-shard may only be enabled in conjunction with query-frontend.cache-results")
```

Comment:
Do you mind defining a `const` for `query-frontend.query-sharding-max-series-per-shard`, so we use it both in the CLI flag definition and here too, please? Same for `query-frontend.cache-results`. Also remember to add a `-` prefix to `query-frontend.cache-results` here.